SPECS/libsolv/libsolv.spec

```diff
@@ -10,6 +10,7 @@ Vendor:         Microsoft Corporation
 Distribution:   Azure Linux
 BuildRequires:  cmake
 BuildRequires:  rpm-devel
+BuildRequires:  libzstd-devel
```

Comment:
I suspect you might need to update the [build_official_toolchain_rpms.sh](https://github.com/microsoft/azurelinux/blob/3.0/toolkit/scripts/toolchain/build_official_toolchain_rpms.sh) file to handle the new dependency between `libsolv` and `libzstd-devel`.

Reply:
Thanks for the call-out; I'm not familiar with this script and how dependencies are encoded there.

I see that it already has `build_rpm_in_chroot_no_install zstd` (right near similar for `gzip`) but it's not clear to me if that's sufficient. @cwize1 would you mind taking a quick look and provide some guidance?

Reply:
That script is basically a manually specified topological ordering of the packages that make up the toolkit. Every package in the toolchain needs to have a `build_rpm_in_chroot_no_install` line. However, for packages that need to be installed to build other packages (i.e. `BuildRequires`), then you need to add a `chroot_and_install_rpms` line for that package as well. We also typically add comments describing what the build dependencies are.

For example, currently there is a line that says:

`# libsolv needs cmake`

This should probably be updated to:

`# libsolv needs cmake, libzstd`

And the following line should be added:

`chroot_and_install_rpms libzstd`

Once you think you have it right, use the `[NoArch-0-OneBranch]-Dev-PrepareBuild` pipeline to test it.

Reply:
And for local build debugging (if the pipeline is going to fail for some reason), you can also download the 3.0 raw toolchain, put it under `build/toolchain/toolchain_from_container.tar.gz`, then run `touch` on it to let `make` know it's fresh and run a local `make toolchain QUICK_REBUILD=y`. Alternatively, you can create this PR against `fasttrack/3.0` and that'll trigger the PR check, which builds the toolchain as the first step anyway and publishes the built RPMs.

Reply:
If `[AMD64-1-OneBranch]-Dev-BuildToolchain` succeeds, then you should be good.